### PR TITLE
Feature/input params as json

### PR DESCRIPTION
**Pull request by olelukoje**

### CODE REVIEW

File: api_spec.yaml [hunk 1]
```diff
@@ -19,11 +19,11 @@ paths:
...
       description: Returns model response based on input parameters
       requestBody:
-        description: File with input parameters
+        description: JSON with input parameters
         content:
-          text/plain:
+          application/json:
             schema:
-              type: string
+              type: object
```
Comment (denisvmedyantsev) [inline api_spec.yaml]: Add details (object fields). Better allow a list of object, batch mode.

File: predicts_api.py [hunk 2]
```diff
@@ -18,18 +17,16 @@ def start():
...
         try:
-            json_path = request.get_data()
-            with open(json_path, "r") as f:
-                json_data = json.load(f)
-            df = pd.DataFrame({0: json_data}).transpose()
+            json_data = request.get_json()
+            if json_data is None:
+                return jsonify("Expected json")
+            else:
+                df = pd.DataFrame({0: json_data}).transpose()
```
Comment (denisvmedyantsev) [inline predicts_api.py]: Model predict method works with array-like input. Is there any reason to use pandas.DataFrame?

**Review by denisvmedyantsev — CHANGES_REQUESTED** [review]

File: api_spec.yaml (hunk 1 above)
Comment (olelukoje) [reply api_spec.yaml]: done

File: predicts_api.py (hunk 2 above)
Comment (olelukoje) [reply predicts_api.py]: done

File: predicts_api.py [hunk 3]
```diff
@@ -18,14 +17,15 @@ def predict():
     if request.method == 'POST':
         try:
             json_data = request.get_json()
+            print(json_data)
             if json_data is None:
                 return jsonify("Expected json")
             else:
-                df = pd.DataFrame({0: json_data}).transpose()
+                data = [list(d.values()) for d in json_data]
```
Comment (denisvmedyantsev) [inline predicts_api.py:23]: Is it safe to just convert dict to an array? What order of fields do you expect? What can you really receive?

**Review by denisvmedyantsev — CHANGES_REQUESTED** [review]

File: predicts_api.py (hunk 3 above)
Comment (olelukoje) [reply predicts_api.py:23]: `d.values()` keeps the same order as it was in dict.

File: api_spec.yaml [hunk 4]
```diff
@@ -23,7 +23,44 @@ paths:
...
+                  total sulfur dioxide:
+                    type: int32
+                  density:
+                    type: number
+                  pH:
+                    type: number
+                  sulphates:
+                    type: number
+                  alcohol:
+                    type: int32
```
Comment (gavrin-s) [inline api_spec.yaml]: Why alcohol type is int32? It is number.

**Review by gavrin-s — CHANGES_REQUESTED** [review]

File: api_spec.yaml (hunk 4 above)
Comment (olelukoje) [reply api_spec.yaml]: I agree, fixed.

**Review by gavrin-s — APPROVED** [review]

File: predicts_api.py [hunk 5]
```diff
@@ -21,11 +20,12 @@ def predict():
...
+                print(data)
```
Comment (denisvmedyantsev) [inline predicts_api.py]: remove debug output

**Review by denisvmedyantsev — APPROVED** [review]

File: predicts_api.py (hunk 5 above)
Comment (olelukoje) [reply predicts_api.py]: done

**Pull request merged**